### PR TITLE
feat: add sentinel bootstrap method

### PR DESCRIPTION
**Pull request by nathanielc**

### CODE REVIEW

File: runner/src/bootstrap.rs [hunk 1]
```diff
@@ -94,3 +97,25 @@ async fn random(n: usize, peers: &BTreeMap<usize, PeerInfo>) -> Result<()> {
     }
     Ok(())
 }
+
+#[tracing::instrument(skip(peers), fields(peers.len = peers.len()))]
+async fn sentinel(n: usize, peers: &BTreeMap<usize, PeerInfo>) -> Result<()> {
+    for peer_info in peers.values() {
+        // Connect to each peer to the first n peers
```
Comment (smrz2001) [inline runner/src/bootstrap.rs:104]: Should this be `Connect each peer to the first n peers`?

File: runner/src/bootstrap.rs [hunk 2]
```diff
@@ -94,3 +97,25 @@ async fn random(n: usize, peers: &BTreeMap<usize, PeerInfo>) -> Result<()> {
...
+        // Do not attempt to connect to more peers than exist.
+        for sentinel in 0..min(n, peers.len() - 1) {
+            debug!(%peer_info.index, %sentinel, "sentinel peer connection");
+            if let Some(sentinel_info) = peers.get(&sentinel) {
```
Comment (smrz2001) [inline runner/src/bootstrap.rs:108]: Does it matter that we might call `connect_peers` with the same peer for both sides of the connection?

**Review by smrz2001 — APPROVED** [review]
Couple of minor questions but otherwise LGTM!

**Review by zachferland — APPROVED** [review]

File: runner/src/bootstrap.rs (hunk 2 above)
Comment (nathanielc) [reply runner/src/bootstrap.rs:108]: It will pollute logs, as the API call will fail. I fixed this in #35 as I had to rewrite bits of it anyway.

**Pull request merged**